streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -246,9 +246,7 @@ class StreamingContext private[streaming] (
     checkpointDir != null
   }
 
-  private[streaming] def initialCheckpoint: Checkpoint = {
-    if (isCheckpointPresent) _cp else null
-  }
+  private[streaming] def initialCheckpoint: Checkpoint = _cp
```

Comment:
This changes the behavior. It could be fine but I don't know if it belongs with typo changes